docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+The [ACL plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin)<sup>[2](#r2)</sup> that is deployed
+as part of aggregated logging will be updated to allow access from the gateway. Additionally, the ElasicSearch index management<sup>[3](#r3)</sup>  will be modified so that it will not cull APIMan data.
+
+
+Cluster administrators can configure APIMan to use an alternate, off cluster ElasticSearch instance if desired.  Some organizations may have an existing instance of ElasticSearch cluster that already defines
+policy, and adminstrators may desire to use it instead of the one provided with aggregated logging.
+
+## Use Cases
+* **UC01** As an API provider, I want to navigate from the OpenShift web console to the policy management interface, so I can define my service policies.
+* **UC02** As an API provider, I want to explicitly expose my service when navigating to the policy managment interface, so that it is consumable according to my management policy.  
+* **UC03** As a cluster administrator, I want a gateway to route API traffic only, so that it manages traffic based on a service's policy.
+* **UC04** As a cluster administrator, I want to deploy APIMan and its components reusing existing infrastructure components where possible (e.g. ElasticSearch), so I can minimize infrastructure components.
+* **UC05** As a cluster administrator, I want to reuse the aggregated logging CA when generating certs for the APIMan components, so I can reuse the ElasticSearch cluster.
+* **UC06** As an API consumer, I want to configure my services to consume other services
+
+## Specification
+
+API services will be scoped to OpenShift projects. Projects will have one-to-one relationship to an APIMan namespace (formally organization). APIMan will utilize namespace
+to manage policy regarding the service.
+
+### Deployment Scenario - APIMan Gateway Fronted by a Router
+
+The primary deployment scenario is to deploy APIMan components as shown in **Figure 1**.  APIMan will be deployed as a cluster level service for managing API service end points. APIMan will be deployed to reuse the existing ElasticSearch cluster for storage.  Communication between APIMan and ElasticSearch will make use of the CA certificate that was used for aggregated logging.  Modifications to the aggregated logging deployer will be made to save the CA as a secret.  This will allow the APIMan deployer to reuse the certificate to create client certificates.
+
+
+**Figure 1 - APIMan Deployment Diagram**
+
+![APIMan Deployment Diagram](apiman_deployment_diagram.png "Figure 1")
+
+The APIMan gateway is deployed in conjunction with a supported OpenShift router.  A typical request to a managed endpoint is initially routed to the gateway and then further routed by the gateway to the targeted service.  This flow can be seen in **Figure 2** where the APIMan gateway relies upon the service proxy to find pods.
```

Comment:
Having the request go through the router and then the APIMan gateway means that we are not using the router for anything except resolving the name to the gateway. Sticky sessions / stick tables / green-blue / etc is not available via APIMan routing via the service proxy.

I would argue that DNS can do the job of routing the name to the APIMan gateway, and the APIMan gateway should be combined in the same pod with an existing HAproxy router to do actual routing after the API management has occurred. This would also provide finer control on where the APIMan gateway can be exposed, by choosing which nodes to place it on and how to expose those nodes.